test/src/main/java/org/apache/accumulo/test/functional/ConcurrentTableNameOperationsIT.java

```diff
@@ -0,0 +1,414 @@
+      assertTrue(client.tableOperations().exists(importTableName),
+          "Expected import table " + importTableName + " to exist");
+      assertEquals(tableCountBefore + 1, client.tableOperations().list().size(),
+          "Expected +1 table count for import operation");
+    }
+
+    pool.shutdown();
+  }
+
+  /**
+   * Test that when several operations all target the same table name, only one operation
+   * successfully creates that table.
+   */
+  @Test
+  public void mixedTableOperations() throws Exception {
+    final int operationsPerType = 10;
+    final int numTasks = operationsPerType * 3;
+    final int numIterations = 3;
+    ExecutorService pool = Executors.newFixedThreadPool(numTasks);
+    String[] expectedTableNames = getUniqueNames(numIterations);
+
+    for (String targetTableName : expectedTableNames) {
+      List<String> cloneSourceTables = new ArrayList<>();
+      List<String> renameSourceTables = new ArrayList<>();
+      for (int i = 0; i < operationsPerType; i++) {
+        String cloneSource = targetTableName + "_clone_src_" + i;
+        client.tableOperations().create(cloneSource);
+        cloneSourceTables.add(cloneSource);
+
+        String renameSource = targetTableName + "_rename_src_" + i;
+        client.tableOperations().create(renameSource);
+        renameSourceTables.add(renameSource);
+      }
+
+      int tableCountBefore = client.tableOperations().list().size();
+
+      List<Future<Boolean>> futures = new ArrayList<>();
+      CountDownLatch startSignal = new CountDownLatch(numTasks);
+      AtomicReference<String> successfulOperation = new AtomicReference<>();
+
+      for (int i = 0; i < operationsPerType; i++) {
+        futures.add(pool.submit(() -> {
+          try {
+            startSignal.countDown();
+            startSignal.await();
+            client.tableOperations().create(targetTableName);
+            successfulOperation.set("create");
+            return true;
+          } catch (TableExistsException e) {
+            return false;
+          }
+        }));
+
+        final int index = i;
+
+        futures.add(pool.submit(() -> {
+          try {
+            startSignal.countDown();
+            startSignal.await();
+            client.tableOperations().rename(renameSourceTables.get(index), targetTableName);
+            successfulOperation.set("rename");
+            return true;
+          } catch (TableExistsException e) {
+            return false;
+          }
+        }));
+
+        futures.add(pool.submit(() -> {
+          try {
+            startSignal.countDown();
+            startSignal.await();
+            client.tableOperations().clone(cloneSourceTables.get(index), targetTableName, true,
+                Map.of(), Set.of());
+            successfulOperation.set("clone");
+            return true;
+          } catch (TableExistsException e) {
+            return false;
+          }
+        }));
+      }
+
```

Comment:
Could add this to check test expectations `assertEquals(numTask, futures.size())`